File: CStoreManagement/app/src/main/java/com/c_store/zhiyazhang/cstoremanagement/model/contractType/ContractTypeModel.java
package com.c_store.zhiyazhang.cstoremanagement.model.contractType;

import com.c_store.zhiyazhang.cstoremanagement.bean.ContractTypeResult;
import com.c_store.zhiyazhang.cstoremanagement.bean.UserBean;
import com.c_store.zhiyazhang.cstoremanagement.utils.callback.MyStringCallBack;
import com.c_store.zhiyazhang.cstoremanagement.utils.url.AppUrl;
import com.google.gson.Gson;
import com.zhy.http.okhttp.OkHttpUtils;

import okhttp3.Call;

/**
 * Created by zhiya.zhang
 * on 2017/5/9 16:50.
 */

public class ContractTypeModel implements ContractTypeInterface {
    @Override
    public void getAllContractType(UserBean user, final ContractTypeListener contractTypeListener) {
        OkHttpUtils
                .get()
                .url(AppUrl.CONTRACT_TYPE_URL)
                .addHeader("Authorization", user.getUid())
                .build()
                .execute(new MyStringCallBack() {
                    @Override
                    public void onError(Call call, Exception e, int id) {
                        super.onError(call, e, id);
                        try {
                            contractTypeListener.contractFailed(myError());
                        } catch (Exception ignored) {
                            if (e.getMessage().contains("failed to connect to")) {
                                contractTypeListener.contractFailed("连接超时");
                            } else {
                                contractTypeListener.contractFailed(e.getMessage());
                            }
                        }
                    }

                    @Override
                    public void onResponse(String response, int id) {
                        super.onResponse(response, id);
                        contractTypeListener.contractSuccess(new Gson().fromJson(response, ContractTypeResult.class));
                    }
                });
    }
}

File: KotlinCStoreManagement/app/src/main/java/com/cstore/zhiyazhang/cstoremanagement/model/contractType/ContractTypeModel.kt
package com.cstore.zhiyazhang.cstoremanagement.model.contractType

import com.cstore.zhiyazhang.cstoremanagement.bean.ContractTypeResult
import com.cstore.zhiyazhang.cstoremanagement.bean.User
import com.cstore.zhiyazhang.cstoremanagement.model.MyListener
import com.cstore.zhiyazhang.cstoremanagement.url.AppUrl
import com.google.gson.Gson
import com.zhiyazhang.mykotlinapplication.utils.MyStringCallBack
import com.zhy.http.okhttp.OkHttpUtils

/**
 * Created by zhiya.zhang
 * on 2017/6/12 15:54.
 */
class ContractTypeModel : ContractTypeInterface {

    override fun getAllContractType(user: User, listener: MyListener) {
        OkHttpUtils
                .get()
                .url(AppUrl.CONTRACT_TYPE_URL)
                .addHeader("Authorization", user.uId)
                .addHeader("store_id", "091209")
                .build()
                .execute(object : MyStringCallBack(listener) {
                    override fun onResponse(response: String, id: Int) {
                        listener.contractSuccess(Gson().fromJson(response, ContractTypeResult::class.java))
                    }
                })
    }
}

interface ContractTypeInterface {
    fun getAllContractType(user: User, listener: MyListener)
}